cardano-node/src/Cardano/CLI/Byron/UpdateProposal.hs

```diff
@@ -0,0 +1,192 @@
+import           Cardano.Config.Types
+import           Cardano.Crypto.Signing (SigningKey, noPassSafeSigner)
+import           Ouroboros.Consensus.Byron.Ledger.Block (ByronBlock)
+import qualified Ouroboros.Consensus.Byron.Ledger.Mempool as Mempool
+
+import           Cardano.CLI.Ops (CliError(..), decodeCBOR, readGenesis)
+
+data ParametersToUpdate =
+    ScriptVersion Word16
+  | SlotDuration Natural
+  | MaxBlockSize Natural
+  | MaxHeaderSize Natural
+  | MaxTxSize Natural
+  | MaxProposalSize Natural
+  | MpcThd LovelacePortion
+  | HeavyDelThd LovelacePortion
+  | UpdateVoteThd LovelacePortion
+  -- UpdateVoteThd: This represents the minimum percentage of the total number of genesis
+  -- keys that have to endorse a protocol version to be able to become adopted.
+  | UpdateProposalThd LovelacePortion
+  -- UpdateProposalTTL: If after the number of slots specified the proposal
+  -- does not reach majority of approvals, the proposal is simply discarded.
+  | UpdateProposalTTL SlotNumber
+  | SoftforkRuleParam SoftforkRule
+  | TxFeePolicy TxFeePolicy
+  | UnlockStakeEpoch EpochNumber
+  deriving Show
+
+createProtocolParametersUpdate
+  :: ProtocolParametersUpdate
+  -> [ParametersToUpdate]
+  -> ProtocolParametersUpdate
+createProtocolParametersUpdate init paramsToUpdate = go init paramsToUpdate
+ where go i [] = i
+       go i (paramToUpdate : rest) =
+         case paramToUpdate of
+           ScriptVersion val -> go i{ppuScriptVersion = Just val} rest
+           SlotDuration val -> go i{ppuSlotDuration = Just val} rest
+           MaxBlockSize val -> go i{ppuMaxBlockSize = Just val} rest
+           MaxHeaderSize val -> go i{ppuMaxHeaderSize = Just val} rest
+           MaxTxSize val -> go i{ppuMaxTxSize = Just val} rest
+           MaxProposalSize val -> go i{ppuMaxProposalSize = Just val} rest
+           MpcThd val -> go i{ppuMpcThd = Just val} rest
+           HeavyDelThd val -> go i{ppuHeavyDelThd = Just val} rest
+           UpdateVoteThd val -> go i{ppuUpdateVoteThd = Just val} rest
+           UpdateProposalThd val -> go i{ppuUpdateProposalThd = Just val} rest
+           UpdateProposalTTL val -> go i{ppuUpdateProposalTTL = Just val} rest
+           SoftforkRuleParam val -> go i{ppuSoftforkRule = Just val} rest
+           TxFeePolicy val -> go i{ppuTxFeePolicy = Just val} rest
+           UnlockStakeEpoch val -> go i{ppuUnlockStakeEpoch = Just val} rest
+
+convertProposalToGenTx :: AProposal ByteString -> Mempool.GenTx ByronBlock
+convertProposalToGenTx prop = Mempool.ByronUpdateProposal (recoverUpId prop) prop
+
+createUpdateProposal
+  :: DbFile
+  -> ConfigYamlFilePath
+  -> SigningKey
+  -> [ParametersToUpdate]
+  -> ExceptT CliError IO (AProposal ByteString)
+createUpdateProposal dbFile configFile sKey paramsToUpdate = do
+
+  latestBlockBS <- getMostRecentBlock dbFile
+
+  latestBlock <- case decodeCBOR latestBlockBS (fromCBORABlockOrBoundary $ EpochSlots 21600) of
```

Comment:
Nitpick, but maybe `21600` could be put into a `where` binding with a name describing what it represents? e.g. `byronEpochSlots` or something.

Or actually, it seems that this same value is exposed from [`Cardano.Chain.Epoch.File` as `mainnetEpochSlots`](https://github.com/input-output-hk/cardano-ledger/blob/a773ba71eb9a6fb91dda2cdf6c41b2280f2acf7d/cardano-ledger/src/Cardano/Chain/Epoch/File.hs#L74-L75). You could use that directly instead.